src/SFA.DAS.EmployerCommitmentsV2.Web.UnitTests/Mappers/Cohort/DetailsViewModelMapperTests.cs

```diff
@@ -6,93 +6,178 @@
-                .ReturnsAsync(_draftApprenticeshipsResponse);
+            var fixture = new DetailsViewModelMapperTestsFixture();
+            var result = await fixture.Map();
+            Assert.AreEqual(fixture.Source.AccountHashedId, result.AccountHashedId);
+        }
 
-            _mapper = new DetailsViewModelMapper(_commitmentsApiClient.Object);
-            _source = autoFixture.Create<DetailsRequest>();
-            _result = await _mapper.Map(TestHelper.Clone(_source));
+        [Test]
+        public async Task WithPartyIsMappedCorrectly()
+        {
+            var fixture = new DetailsViewModelMapperTestsFixture();
+            var result = await fixture.Map();
+            Assert.AreEqual(fixture.Cohort.WithParty, result.WithParty);
         }
 
         [Test]
-        public void AccountHashedIdIsMappedCorrectly()
+        public async Task LegalEntityNameIsMappedCorrectly()
         {
-            Assert.AreEqual(_source.AccountHashedId, _result.AccountHashedId);
+            var fixture = new DetailsViewModelMapperTestsFixture();
+            var result = await fixture.Map();
+            Assert.AreEqual(fixture.Cohort.LegalEntityName, result.LegalEntityName);
         }
 
         [Test]
-        public void WithPartyIsMappedCorrectly()
+        public async Task ProviderNameIsMappedCorrectly()
         {
-            Assert.AreEqual(_cohort.WithParty, _result.WithParty);
+            var fixture = new DetailsViewModelMapperTestsFixture();
+            var result = await fixture.Map();
+            Assert.AreEqual(fixture.Cohort.ProviderName, result.ProviderName);
         }
 
         [Test]
-        public void LegalEntityNameIsMappedCorrectly()
+        public async Task MessageIsMappedCorrectly()
         {
-            Assert.AreEqual(_cohort.LegalEntityName,_result.LegalEntityName);
+            var fixture = new DetailsViewModelMapperTestsFixture();
+            var result = await fixture.Map();
+            Assert.AreEqual(fixture.Cohort.LatestMessageCreatedByProvider, result.Message);
         }
 
         [Test]
-        public void ProviderNameIsMappedCorrectly()
+        public async Task CohortReferenceIsMappedCorrectly()
         {
-            Assert.AreEqual(_cohort.ProviderName, _result.ProviderName);
+            var fixture = new DetailsViewModelMapperTestsFixture();
+            var result = await fixture.Map();
+            Assert.AreEqual(fixture.Source.CohortReference, result.CohortReference);
         }
 
         [Test]
-        public void MessageIsMappedCorrectly()
+        public async Task TransferSenderHashedIdIsEncodedCorrectlyWhenThereIsAValue()
         {
-            Assert.AreEqual(_cohort.LatestMessageCreatedByProvider, _result.Message);
+            var fixture = new DetailsViewModelMapperTestsFixture().SetTransferSenderId(123);
+            var result = await fixture.Map();
+            Assert.AreEqual("X123X", result.TransferSenderHashedId);
```

Comment:
Just a minor observation - it's not obvious where the expected value comes from - a quick inspection soon reveals the setup on the encoding service is coded to do this. I wonder whether it might be a little clearer if the expected value was passed into the SetTransferSenderId as a second parameter? The setup on the encoding service could be done there, which would also restrict the encoder to the expected values (rather than accepting any long as it currently does).

Reply:
good idea. will do